x/attribute/keeper/keeper.go

```diff
@@ -160,6 +162,36 @@ func (k Keeper) SetAttribute(
 	return nil
 }
 
+// IncAddNameAddressLookup increments the count of name to address lookups
+func (k Keeper) IncAddNameAddressLookup(ctx sdk.Context, attr types.Attribute) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.AttributeNameAddrKeyPrefix(attr.Name, attr.GetAddressBytes())
+	bz := store.Get(key)
+	id := uint64(0)
+	if bz != nil {
+		id = binary.BigEndian.Uint64(bz)
+	}
+	bz = sdk.Uint64ToBigEndian(id + 1)
+	store.Set(key, bz)
+}
+
+// DecAddNameAddressLookup decrements the name to account lookups and removes value if decremented to 0
+func (k Keeper) DecAddNameAddressLookup(ctx sdk.Context, attr types.Attribute) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.AttributeNameAddrKeyPrefix(attr.Name, attr.GetAddressBytes())
+	bz := store.Get(key)
+	id := uint64(0)
+	if bz != nil {
+		value := binary.BigEndian.Uint64(bz)
+		if value <= uint64(1) {
+			store.Delete(key)
+		} else {
+			id = binary.BigEndian.Uint64(bz) - uint64(1)
+			store.Set(key, sdk.Uint64ToBigEndian(id-1))
```

Comment:
Are you purposely double decrementing? Also you may want to use value instead of `binary.BigEndian.Uint64(bz)`. If you have a value of 2 then id will be 1, and you will be setting the value to 0.